R/compare.R

```diff
@@ -69,41 +61,68 @@ compare_output <- function(model, pars_obs, data, type="sircovid_basic") {
       ## sum model ITU cases output across ages/infectivities
       model_icu <- colSums(state[index_ICU, ])
       log_weights <- log_weights +
-        ll_nbinom(data$itu[t], model_icu, phi_ICU, k_ICU, exp_noise)
+        ll_nbinom(data$itu[t], phi_ICU * model_icu, k_ICU, exp_noise)
     }
     
-    if (type %in% c("sircovid_hospital","sircovid_serology")  && !is.na(data$general[t])) {
+    if (type %in% c("sircovid_hospital","sircovid_serology","sircovid_serology2")  && !is.na(data$general[t])) {
       ## sum model output across ages/infectivities
       model_general <- colSums(state[index_general, ])
       log_weights <- log_weights +
-        ll_nbinom(data$general[t], model_general, phi_general, k_general, exp_noise)
+        ll_nbinom(data$general[t], phi_general * model_general, k_general, exp_noise)
     }
-
-    if (!is.na(data$deaths[t])) {
-      ## new deaths summed across ages/infectivities
-      model_deaths <- colSums(state[index_D, ]) -
-        colSums(prev_state[index_D, ])
-      log_weights <- log_weights +
-        ll_nbinom(data$deaths[t], model_deaths, phi_death, k_death, exp_noise)
+    
+    if (type!="sircovid_serology2"){
+      if (!is.na(data$deaths[t])) {
+        ## new deaths summed across ages/infectivities
+        model_deaths <- colSums(state[index_D, ]) -
+          colSums(prev_state[index_D, ])
+        log_weights <- log_weights +
+          ll_nbinom(data$deaths[t], phi_death * model_deaths, k_death, exp_noise)
+      }
+    } else {
+      if (!is.na(data$deaths[t])) {
+        ## new deaths summed across ages/infectivities
+        model_deaths_hosp <- colSums(state[index_D_hosp, ]) -
```

Comment:
I think this would need extra if clauses (because we only need to do this if deaths/deaths_hosp/deaths_comm are NA as appropriate). So think I'll just leave this as is.